### ISSUE TITLE
Invoke-Command does not raise Verbose/Debug/Warning Stream events from Microsoft.PowerShell.SDK 7.x

### ISSUE THREAD

**Issue by hoppi1804** [PowerShell/PowerShell#14843]
It seems the SDK will not raise any events from Debug/Verbose/Warning Streams, when called remotely by "Invoke-Command". Events for Error and Information streams are raised without any issues.

Please see the repro. Maybe I am overlooking something obvious?

## Steps to reproduce

```c#
using System;
using System.Management.Automation;
using System.Management.Automation.Runspaces;
using System.Threading.Tasks;
using Microsoft.PowerShell;

namespace PowerShellRepro
{
    public class PowerShellService
    {
        private void General_DataAdding(object sender, DataAddingEventArgs e)
        {
            switch (e.ItemAdded)
            {
                case VerboseRecord verboseRecord:
                    Console.WriteLine($"Verbose: {verboseRecord.Message}");
                    break;
                case DebugRecord debugRecord:
                    Console.WriteLine($"Debug:   {debugRecord.Message}");
                    break;
                case ErrorRecord errorRecord:
                    Console.WriteLine($"Error:   {errorRecord.Exception.Message}");
                    break;
                case InformationRecord informationRecord:
                    Console.WriteLine($"Info:    {informationRecord.MessageData}");
                    break;
            }
        }

        public async Task RunTestAsync(string remoteSystem)
        {
            var iss = InitialSessionState.CreateDefault();
            iss.ExecutionPolicy = ExecutionPolicy.Unrestricted;
            using var runspace = RunspaceFactory.CreateRunspace(iss);

            runspace.Open();
            runspace.SessionStateProxy.SetVariable("VerbosePreference", "Continue");
            runspace.SessionStateProxy.SetVariable("DebugPreference", "Continue");

            using var poshSession = System.Management.Automation.PowerShell.Create();


            poshSession.Streams.Verbose.DataAdding += General_DataAdding;
            poshSession.Streams.Debug.DataAdding += General_DataAdding;
            poshSession.Streams.Error.DataAdding += General_DataAdding;
            poshSession.Streams.Information.DataAdding += General_DataAdding;

            poshSession.Runspace = runspace;

            var psScript = @"

[CmdletBinding()]
param(
    [string] $computerName
)

Write-Verbose ""[$($env:ComputerName)] Local - Test before - Remote System [$($computerName)]""
Write-Debug   ""[$($env:ComputerName)] Local - Posh Version: $($PsVersionTable.PSVersion)""

Invoke-Command -ComputerName $computerName -ScriptBlock { 
    [CmdletBinding()]
    param() 
    
    Write-Information -MessageData ""[$env:ComputerName] Remote - Test Informational before"";
    $InformationPreference = ""Continue"";
    Write-Information -MessageData ""[$env:ComputerName] Remote - Test Informational after"";
    Write-Information ""[$($env:ComputerName)] Remote - Posh Version: $($PsVersionTable.PSVersion)"";

    Write-Error ""[$($env:ComputerName)] Remote - Test Error"";

    Write-Verbose ""[$($env:ComputerName)] Remote - Test Verbose before""; 
    $VerbosePreference = ""Continue""; 
    Write-Verbose ""[$($env:ComputerName)] Remote - Test Verbose after"";

    Write-Debug ""[$($env:ComputerName)] Remote - Test Debug before""; 
    $DebugPreference = ""Continue""; 
    Write-Debug ""[$($env:ComputerName)] Remote - Test Debug after"";
};
Write-Verbose ""[$($env:ComputerName)] Local - Test after""

";

            poshSession.Commands.AddScript(psScript);
            poshSession.AddParameter("ComputerName", remoteSystem);
            //poshSession.AddParameter("Verbose", true);
            //poshSession.AddParameter("Debug", true);

            await poshSession.InvokeAsync().ConfigureAwait(false);
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            var defaultRemoteSystem = "<somesystemname>";

            Console.WriteLine($"Please enter servername [{defaultRemoteSystem}]");

            var remoteSystem = Console.ReadLine();
            if (string.IsNullOrEmpty(remoteSystem))
            {
                remoteSystem = defaultRemoteSystem;
            }

            var ps = new PowerShellService();

            ps.RunTestAsync(remoteSystem).Wait();
        }
    }
}
```

## Expected behavior

```none
Please enter servername [<somesystemname>]
shph01
Verbose: [<localSystem>] Local - Test before - Remote System [shph01]
Debug:   [<localSystem>] Local - Posh Version: 7.1.2
Info:    [<somesystemname>] Remote - Test Informational before
Info:    [<somesystemname>] Remote - Test Informational after
Info:    [<somesystemname>] Remote - Posh Version: 5.1.17763.1490
Error:   [<somesystemname>] Remote - Test Error
Verbose:   [<somesystemname>] Remote - Test Verbose before
Verbose:   [<somesystemname>] Remote - Test Verbose after
Debug:   [<somesystemname>] Remote - Test Debug before
Debug:   [<somesystemname>] Remote - Test Debug after
Verbose: [<localSystem>] Local - Test after
```

## Actual behavior

```none
Please enter servername [<somesystemname>]
shph01
Verbose: [<localSystem>] Local - Test before - Remote System [shph01]
Debug:   [<localSystem>] Local - Posh Version: 7.1.2
Info:    [<somesystemname>] Remote - Test Informational before
Info:    [<somesystemname>] Remote - Test Informational after
Info:    [<somesystemname>] Remote - Posh Version: 5.1.17763.1490
Error:   [<somesystemname>] Remote - Test Error
Verbose: [<localSystem>] Local - Test after
```

## Environment data

local System: 7.0.x / 7.1.x
remote System: 5.1.17763
```none

```

**Comment by SteveL-MSFT** [comment]
@hoppi1804 can you confirm if this worked with Windows PowerShell 5.1?  Would be good to know if this is a regression.  Thanks.

**Comment by hoppi1804** [comment]
No, with Microsoft.PowerShell.5.1.ReferenceAssemblies v1.0.0 this is not working, either.

**Comment by SteveL-MSFT** [comment]
This seems to be "by design" if it doesn't work in 5.1.  cc @PaulHigin if he has any insights

**Comment by PaulHigin** [comment]
I see this behavior too, and it looks like a bug to me.
```powershell
$ps.AddScript('icm . { write-verbose "hello" -verbose }').Invoke()
$ps.Streams

Error       : {}
Progress    : {}
Verbose     : {}
Debug       : {}
Warning     : {}
Information : {}
```

It looks like Invoke-Command is not writing to the verbose, debug streams.  But it is strange that informational data stream works.
My guess is that Invoke-Command is calling the client remote host method for verbose/debug data and bypassing the data stream handling.  There was an issue about this years ago but I don't remember the details.

**Comment by PeteBoyRocket** [comment]
I _think_ I'm seeing the same issue, where my Write-Warning calls seem to ignore ` -WarningAction Continue` when used in a Job in a `ForEach-Object -Parallel` loop - please can someone confirm? If not I'll raise a separate issue:

```
$InformationPreference = 'Continue'
Write-Information "Script start. InformationPreference = $InformationPreference, WarningPreference = $WarningPreference, ErrorActionPreference = $ErrorActionPreference"

Write-Information "Starting test NOT in parallel loop..."

$notInParallelLoopJobScriptBlock = [scriptblock]::Create(
    {
        param($tenant) 
        Write-Information "Started job for $tenant. InformationPreference = $InformationPreference, WarningPreference = $WarningPreference, ErrorActionPreference = $ErrorActionPreference"
        Start-Sleep -Seconds (Get-Random -Minimum 1 -Maximum 5)
        
        Write-Warning "Warning here! $tenant"
        Write-Warning "Warning here forcing warningAction to continue! $tenant" -WarningAction Continue
        Write-Information "Finished job for $tenant" 
    })

$notInParallelLoopJob = Start-Job $notInParallelLoopJobScriptBlock -ArgumentList 'a'

Wait-Job $notInParallelLoopJob
Receive-Job $notInParallelLoopJob

Write-Information "Finished test NOT in parallel loop."
Write-Information "Starting test IN parallel loop..."

$tenants = 'b'

$tenants | ForEach-Object -Parallel {
    Write-Information "Updating $_"

    $inParallelLoopJobScriptBlock = [scriptblock]::Create(
        {
            param($tenant) 
            Write-Information "Started job for $tenant. InformationPreference = $InformationPreference, WarningPreference = $WarningPreference, ErrorActionPreference = $ErrorActionPreference"
            Start-Sleep -Seconds (Get-Random -Minimum 1 -Maximum 5)
            
            Write-Warning "Warning here! $tenant"
            Write-Warning "Warning here! $tenant" -WarningAction Continue
            Write-Information "Finished job for $tenant" 
        })

    $inParallelLoopJob = Start-Job $inParallelLoopJobScriptBlock -ArgumentList $_
    
    Wait-Job $inParallelLoopJob
    Receive-Job $inParallelLoopJob
    
    Write-Information "Finished $_"
}

Write-Information "Finished test IN parallel loop."
Write-Information 'Script end'
```

Results in:
```
Script start. InformationPreference = Continue, WarningPreference = Continue, ErrorActionPreference = Stop
Starting test NOT in parallel loop...

Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
--     ----            -------------   -----         -----------     --------             -------
238    Job238          BackgroundJob   Completed     True            localhost            …
Started job for a. InformationPreference = SilentlyContinue, WarningPreference = Continue, ErrorActionPreference = Continue
WARNING: Warning here! a
WARNING: Warning here forcing warningAction to continue! a
Finished job for a
Finished test NOT in parallel loop.
Starting test IN parallel loop...
Updating b
240    Job240          BackgroundJob   Completed     False           localhost            …
Started job for b. InformationPreference = SilentlyContinue, WarningPreference = Continue, ErrorActionPreference = Continue
Finished job for b
Finished b
Finished test IN parallel loop.
Script end
```
